components/google_sheets/actions/add-single-row/add-single-row.mjs

```diff
@@ -56,59 +33,48 @@ export default {
           driveId: googleSheets.methods.getDriveId(c.drive),
         }),
       ],
+      description: "",
+      withLabel: true,
     },
     sheetName: {
       propDefinition: [
         googleSheets,
         "sheetName",
         (c) => ({
-          sheetId: c.sheetId,
+          sheetId: c.sheetId.value,
         }),
       ],
     },
-    dataFormat,
+    headerProp,
   },
   async additionalProps() {
     const props = {};
-    if (this.dataFormat == "column") {
-      props.header = headerProp;
-      if (this.header === "hasHeaders") {
-        const { values } = await this.googleSheets.getSpreadsheetValues(this.sheetId, `${this.sheetName}!1:1`);
-        for (let i = 0; i < values[0]?.length; i++) {
-          props[`col_${i.toString().padStart(4, "0")}`] = {
-            type: "string",
-            label: values[0][i],
-            optional: true,
-          };
-        }
-      } else if (this.header === "noHeaders") {
-        props.myColumnData = {
-          type: "string[]",
-          label: "Row Data",
-          description: "Provide a value for each cell of the row. Google Sheets accepts strings, numbers and boolean values for each cell. To set a cell to an empty value, pass an empty string.",
+    if (this.headerProp === "Yes") {
+      const { values } = await this.googleSheets.getSpreadsheetValues(this.sheetId.value, `${this.sheetName}!1:1`);
+      for (let i = 0; i < values[0]?.length; i++) {
```

Comment:
I know this looks just as verbose but this is the way I would approach with Node 14 features:

```javascript
for (const [i, label] of (values[0] ?? []).entries()) {
  props[`col_${i.toString().padStart(4, "0")}`] = {
    type: "string",
    label,
    optional: true,
  };
}
```